### PR TITLE
Emit response event from file#createReadStream()

### PR DESCRIPTION
**Pull request by robertdimarco**
See #606

### CODE REVIEW

File: test/storage/file.js [hunk 1]
```diff
@@ -370,6 +370,30 @@ describe('File', function() {
...
+
+        stream.Readable.call(this);
+        this._read = function() {
+          this.push(data);
+          this.push(null);
+        };
+
+        setImmediate(function() {
+          that.emit('response', fakeResponse);
+          that.emit('complete', fakeResponse);
```
Comment (ryanseys) [inline test/storage/file.js]: We should probably remove response from the `complete` at sometime then you think?

**Comment by ryanseys** [comment]
Can you please document this functionality in the docs for `file#createReadStream` :)

**Comment by robertdimarco** [comment]
@ryanseys I added a line to the `file#createReadStream` header docs but let me know if you'd like to see more there or there's something I need to do to autogenerate-then-commit those docs.

File: test/storage/file.js (hunk 1 above)
Comment (robertdimarco) [inline test/storage/file.js]: Maybe... but I suspect it isn't worth it. People seem to love the `request` module, so perhaps sticking to its conventions isn't a bad thing. =)

Comment (ryanseys) [inline test/storage/file.js]: Oh is that a convention in the `request` module? I thought we just added it! :)

Comment (robertdimarco) [inline test/storage/file.js]: Yeah, you're in good company!

**Comment by ryanseys** [comment]
Thanks @robertdimarco ! :dancer:

**Pull request merged**